rust-connector-sdk/src/connector.rs

```diff
@@ -201,23 +191,15 @@ pub enum MutationError {
 /// connection string would be state.
 #[async_trait]
 pub trait Connector {
-    /// The type of unvalidated, raw configuration, as provided by the user.
-    type RawConfiguration;
     /// The type of validated configuration
     type Configuration;
     /// The type of unserializable state
     type State;
 
-    fn make_empty_configuration() -> Self::RawConfiguration;
-
-    async fn update_configuration(
-        config: Self::RawConfiguration,
-    ) -> Result<Self::RawConfiguration, UpdateConfigurationError>;
-
     /// Validate the raw configuration provided by the user,
     /// returning a configuration error or a validated [`Connector::Configuration`].
     async fn validate_raw_configuration(
-        configuration: Self::RawConfiguration,
+        configuration_dir: PathBuf,
```

Comment:
I suggest weakening this to `AsRef<Path>` so we can be a little more flexible with the input.

```suggestion
        configuration_dir: impl AsRef<Path>,
```